cas-server-core/src/main/java/org/jasig/cas/services/JsonServiceRegistryConfigWatcher.java

```diff
@@ -0,0 +1,188 @@
+                        Reset the key -- this step is critical to receive
+                        further watch events. If the key is no longer valid, the directory
+                        is inaccessible so exit the loop.
+                     */
+                    final boolean valid = (key != null && key.reset());
+                    if (!valid) {
+                        LOGGER.warn("Directory key is no longer valid. Quitting watcher service");
+                        break;
+                    }
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Handle event.
+     *
+     * @param key the key
+     */
+    private void handleEvent(final WatchKey key) {
+        this.readLock.lock();
+        try {
+            for (final WatchEvent<?> event : key.pollEvents()) {
+                if (event.count() <= 1) {
+                    final WatchEvent.Kind kind = event.kind();
+
+                    //The filename is the context of the event.
+                    final WatchEvent<Path> ev = (WatchEvent<Path>) event;
+                    final Path filename = ev.context();
+
+                    final Path parent = (Path) key.watchable();
+                    final Path fullPath = parent.resolve(filename);
+                    final File file = fullPath.toFile();
+
+                    LOGGER.trace("Detected event [{}] on file [{}]. Loading change...", kind, file);
+                    if (kind.name().equals(ENTRY_CREATE.name()) && file.exists()) {
+                        handleCreateEvent(file);
+                    } else if (kind.name().equals(ENTRY_DELETE.name())) {
+                        handleDeleteEvent();
+                    } else if (kind.name().equals(ENTRY_MODIFY.name()) && file.exists()) {
+                        handleModifyEvent(file);
+                    }
+                }
+
+            }
+        } finally {
+            this.readLock.unlock();
+        }
+    }
+
+    /**
+     * Handle modify event.
+     *
+     * @param file the file
+     */
+    private void handleModifyEvent(final File file) {
+    /*
+        load the entry and save it back to the map
+        without any warnings on duplicate ids.
+     */
+        final RegisteredService newService = this.serviceRegistryDao.loadRegisteredServiceFromFile(file);
+        if (newService == null) {
+            LOGGER.warn("New service definition could not be loaded from [{}]", file.getAbsolutePath());
+        } else {
+            final RegisteredService oldService = this.serviceRegistryDao.findServiceById(newService.getId());
+
+            if (!newService.equals(oldService)) {
+                this.serviceRegistryDao.updateRegisteredService(newService);
+                this.serviceRegistryDao.refreshServicesManager();
+            }
+        }
+    }
+
+    /**
+     * Handle delete event.
+     */
+    private void handleDeleteEvent() {
+        this.serviceRegistryDao.load();
+        this.serviceRegistryDao.refreshServicesManager();
```

Comment:
Sadly no, because a DAO is managed by the services manager instance. DAO loads the changes internally, but the manager sort of "caches" them, so you need to refresh both.